### PR TITLE
small doc updates to add auth and fix bg event and theming

### PR DESCRIPTION
**Pull request by tgberkeley**

### CODE REVIEW

File: docs/events/background_events.md [hunk 1]
```diff
@@ -25,6 +25,10 @@ In the following example, the `my_task` event handler is decorated with
 long as certain conditions are met. While it is running, the UI remains
 interactive and continues to process events normally.
 
+```md alert info
+# Background events are similar to Task Queues in [Celery](https://www.fullstackpython.com/celery.html) allowing asynchronous events.
```
Comment (picklelo) [inline docs/events/background_events.md:29]: I don't think this is true. Background events can be used to start task queues etc. but we don't offer all the features of a task queue.

File: docs/styling/theming.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ from pcweb.pages.docs import library
 
 # Theming
 
-As of Reflex `v0.4.0`, you can now theme your Reflex applications. The core of our theming system is directly based on the [Radix Themes](https://radix-ui.com/docs/themes) library. This allows you to easily change the theme of your application along with providing a default light and dark theme.
+As of Reflex `v0.4.0`, you can now theme your Reflex applications. The core of our theming system is directly based on the [Radix Themes](https://www.radix-ui.com) library. This allows you to easily change the theme of your application along with providing a default light and dark theme.
```
Comment (picklelo) [inline docs/styling/theming.md:8]: Maybe specify in the last sentence what theme actually means, all the components will have a unified color appearance

File: docs/styling/theming.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ Additionally you can modify the theme of your app through using the `Theme Panel
 
 ### Color Scheme
 
-On a high-level, component `color_scheme` inherits from the color specified in the theme. This means that if you change the theme, the color of the component will also change.
+On a high-level, component `color_scheme` inherits from the color specified in the theme. This means that if you change the theme, the color of the component will also change. The selection of available colors can be found [here](https://www.radix-ui.com/colors).
```
Comment (picklelo) [inline docs/styling/theming.md:28]: nit: Just say "The available colors"

**Review by picklelo — CHANGES_REQUESTED** [review]
Thanks for adding this - just left some nit comments for future docs in general. Try to always use the simplest language possible when writing the docs. I'll try to make an official style guide so these can be more codified instead of just opinions, but want to have consistent tone throughout the docs.

File: docs/authentication/authentication_overview.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
+# Authentication Overview
+
+Almost any non-trivial web app needs a way to identify and authenticate users,
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/authentication/authentication_overview.md:4]: Done

File: docs/authentication/authentication_overview.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+We have four solutions here:
+
+1. Local Auth: Check out how it is set up in this app: https://github.com/reflex-dev/reflex-examples/tree/main/twitter
+2. Google Auth: Check out how it is set up with this blog: https://reflex.dev/blog/2023-10-25-implementing-sign-in-with-google/
+3. Captcha: Check out this repo: https://github.com/masenf/reflex-google-recaptcha-v2
+4. Magic Link Auth: Check out this repo: https://github.com/masenf/reflex-magic-link-auth
+
+
+There is more thorough authentication documentation on the way. Check back soon!
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/authentication/authentication_overview.md:18]: changed

File: docs/authentication/authentication_overview.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+Almost any non-trivial web app needs a way to identify and authenticate users,
+but Reflex does not provide this functionality out of the box because there are
+way too many different ways to approach the problem. Thanks to the plethora of
+existing React components for performing auth, a wrapper can be created to
+include most third-party auth solutions within a Reflex app.
+
+We have four solutions here:
+
+1. Local Auth: Check out how it is set up in this app: https://github.com/reflex-dev/reflex-examples/tree/main/twitter
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/authentication/authentication_overview.md:12]: changed

File: docs/events/background_events.md (hunk 1 above)
Comment (tgberkeley) [reply docs/events/background_events.md:29]: rephrased

File: docs/styling/theming.md (hunk 3 above)
Comment (tgberkeley) [reply docs/styling/theming.md:28]: changed

**Review by picklelo — APPROVED** [review]

**Pull request merged**